app/models/assignment.rb

```diff
@@ -39,7 +39,8 @@ class Assignment < ApplicationRecord
 
   validate :uniqueness_of_slug_across_organization
   validate :starter_code_repository_not_empty, if: :will_save_change_to_starter_code_repo_id?
-  validate :starter_code_repository_is_template, if: :will_save_change_to_starter_code_repo_id?
+  validate :starter_code_repository_is_template,
+    if: -> { :will_save_change_to_starter_code_repo_id? || :will_save_change_to_template_repos_enabled }
```

Comment:
The code here works good and I'm comfortable moving forward with it. I wonder, though, based on your comment if we should instead only check when the option is enabled and the value is not blank?. With this code if someone points to a third party template and the owner unchecks the template box... it is going to be very difficult to change this.

Reply:
Ah, I misunderstood and can clearly see this checked in the concern: https://github.com/education/classroom/blob/0ec9560df085de5b341d857b6ae1c901097f4991/app/models/concerns/starter_code_importable.rb#L30 👏